src/CollapsibleField.js

```diff
@@ -241,35 +282,60 @@ class CollapsibleField extends Component {
 
   render() {
     let {
-      schema: { title },
+      schema: { title, properties = {} },
       uiSchema,
       registry: { fields },
       idSchema: { $id } = {},
       name,
-      formContext
+      formContext,
+      formData,
+      onChange: propsOnChange
     } = this.props;
     let { collapsed, AddElement } = this.state;
-    let { collapse: { field } } = uiSchema;
+    let {
+      collapse: { collapsibleHeaderElements = { items: {} }, field }
+    } = uiSchema;
     let CollapseElement = fields[field];
     // uischema retains the value form the state
     uiSchema.collapse.collapsed = this.state.collapsed;
 
     title = uiSchema["ui:title"] ? uiSchema["ui:title"] : title ? title : name;
     let customizedId = collapsed ? $id : undefined;
+
+    //remove header elements from the schema
+    let headerElementsSchemas = {};
+    let propertiesNoHeader = { ...properties };
+    Object.keys(collapsibleHeaderElements.items).map(key => {
+      if (propertiesNoHeader[key]) {
+        headerElementsSchemas[key] = propertiesNoHeader[key];
+        delete propertiesNoHeader[key];
+      }
+    });
+
+    const collapseElementProps = {
+      ...this.props,
+      schema: { ...this.props.schema, properties: propertiesNoHeader }
+    };
+
     return (
       <div id={customizedId}>
         <CollapseMenu
+          headerElementsSchemas={headerElementsSchemas}
+          collapsibleFieldId={$id}
           title={title}
           uiSchema={uiSchema}
           collapsed={collapsed}
           formContext={formContext}
+          formData={formData}
+          fields={fields}
           onAdd={this.handleAdd}
           onChange={this.handleCollapsed}
+          propsOnChange={propsOnChange}
         />
         <div className="form-group">
           {AddElement && <AddElement />}
           {!collapsed && <CollapseLegend {...this.props} />}
-          {!collapsed && <CollapseElement {...this.props} />}
+          {!collapsed && <CollapseElement {...collapseElementProps} />}
```

Comment:
Please make sure , it will not affect existing collapsible field.

Reply:
Confirmed that existing collapsible field works as expected